web/src/invite.ts

```diff
@@ -360,6 +388,12 @@ function open_invite_user_modal(e: JQuery.ClickEvent<Document, undefined>): void
             stream_pill_widget = invite_stream_picker_pill.create($stream_pill_container);
         }
 
+        $("#invite_as, #invite_streams_container .input").on(
+            "change",
+            update_guest_visible_users_count,
+        );
+        update_guest_visible_users_count();
```

Comment:
Does the notice hide when the role is changed from "Guest" to something else? I do not see any code which does this.